SoftLayer/managers/ordering.py

```diff
@@ -209,15 +213,250 @@ def get_package_by_key(self, package_keyname, mask=None):
+
+        categories = self.package_svc.getConfiguration(id=package['id'], **get_kwargs)
+        return categories
+
+    def list_items(self, package_keyname, **kwargs):
+        """List the items for the given package.
+
+        :param str package_keyname: The package for which to get the items.
+        :returns: List of items in the package
+
+        """
+        get_kwargs = {}
+        default_mask = '''id,
+                          keyName,
+                          description
+                       '''
+        get_kwargs['mask'] = kwargs.get('mask', default_mask)
+
+        if 'filter' in kwargs:
+            get_kwargs['filter'] = kwargs['filter']
+
+        package = self.get_package_by_key(package_keyname, mask='id')
+        if not package:
+            raise exceptions.SoftLayerError("Package {} does not exist".format(package_keyname))
+
+        items = self.package_svc.getItems(id=package['id'], **get_kwargs)
+        return items
+
+    def list_packages(self, **kwargs):
+        """List active packages.
+
+        :returns: List of active packages.
+
+        """
+        get_kwargs = {}
+        default_mask = '''id,
+                          name,
+                          keyName,
+                          isActive
+                       '''
+        get_kwargs['mask'] = kwargs.get('mask', default_mask)
+
+        if 'filter' in kwargs:
+            get_kwargs['filter'] = kwargs['filter']
+
+        packages = self.package_svc.getAllObjects(**get_kwargs)
+
+        return [package for package in packages if package['isActive']]
+
+    def list_presets(self, package_keyname, **kwargs):
+        """Gets active presets for the given package.
+
+        :param str package_keyname: The package for which to get presets
+        :returns: A list of package presets that can be used for ordering
+
+        """
+        get_kwargs = {}
+        default_mask = '''id,
+                          name,
+                          keyName,
+                          description
+                       '''
+        get_kwargs['mask'] = kwargs.get('mask', default_mask)
+
+        if 'filter' in kwargs:
+            get_kwargs['filter'] = kwargs['filter']
+
+        package = self.get_package_by_key(package_keyname, mask='id')
+        if not package:
+            raise exceptions.SoftLayerError("Package {} does not exist".format(package_keyname))
+
+        acc_presets = self.package_svc.getAccountRestrictedActivePresets(
+            id=package['id'], **get_kwargs)
+        active_presets = self.package_svc.getActivePresets(id=package['id'], **get_kwargs)
+        return acc_presets + active_presets
+
+    def get_preset_by_key(self, package_keyname, preset_keyname, mask=None):
+        """Gets a single preset with the given key."""
+        preset_operation = '_= %s' % preset_keyname
+        _filter = {'activePresets': {'keyName': {'operation': preset_operation}}}
```

Comment:
This is only looking in the active presets. It also needs to look at the account restricted active presets with this filter to possibly find it in there as well

Reply:
turns out this filter works with both activePresets and accountRestrictedActivePresets, so nothing needs to be updated.